syntax = "proto3";

package master;

service Master {
  // Sends a greeting
  rpc getVariable (VariableName) returns (VariableValue) {}
  rpc traceVariable (VariableName) returns (stream VariableValue) {}
  rpc uploadFile (stream Chunk) returns (UploadStatus) {}
}

// The request message containing variable's name.
message VariableName {
	string name = 1;
}

// The response message containing variable's value
message VariableValue {
  	uint64 value = 1;
	uint64 timestamp = 2;
}

message Chunk {
        bytes data = 1;
}

message UploadStatus {
        string msg = 1;
        bool result = 2;
}

message FileInfo {
        string filename = 1;
}
